spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestBaseWithCatalog.java

```diff
@@ -36,16 +41,46 @@
 import org.apache.iceberg.catalog.SupportsNamespaces;
 import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.hadoop.HadoopCatalog;
+import org.apache.iceberg.inmemory.InMemoryCatalog;
+import org.apache.iceberg.rest.RCKUtils;
+import org.apache.iceberg.rest.RESTCatalog;
+import org.apache.iceberg.rest.RESTCatalogServer;
+import org.apache.iceberg.rest.RESTServerExtension;
 import org.apache.iceberg.util.PropertyUtil;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
 import org.junit.jupiter.api.io.TempDir;
 
 @ExtendWith(ParameterizedTestExtension.class)
 public abstract class TestBaseWithCatalog extends TestBase {
-  protected static File warehouse = null;
+  protected static File warehouse;
+
+  static {
+    try {
+      warehouse = File.createTempFile("warehouse", null);
+      assertThat(warehouse.delete()).isTrue();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @RegisterExtension
+  private static RESTServerExtension restServerExtension =
+      new RESTServerExtension(
+          Map.of(
+              RESTCatalogServer.REST_PORT, String.valueOf(RCKUtils.findFreePort()),
+              CatalogProperties.WAREHOUSE_LOCATION, warehouse.getAbsolutePath(),
```

Comment:
I commented on this yesterday but can't find the comment anymore. The `warehouse` is currently only used for Hadoop and the REST server will create a separate temp directory for the REST catalog's warehouse, so we can remove this